internal/discord/command-handlers.go

```diff
@@ -0,0 +1,314 @@
+			Name:        "user",
+			Description: "User being unbanned (Discord ID)",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for unban",
+			Required:    true,
+		},
+	},
+}
+
+var RemoveNicknameCommand = &discordgo.ApplicationCommand{
+	Name:                     "removenickname",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Remove the nickname of the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User whose nickname to remove",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for nickname removal",
+			Required:    true,
+		},
+	},
+}
+
+var SetNicknameCommand = &discordgo.ApplicationCommand{
+	Name:                     "setnickname",
+	DefaultMemberPermissions: &mutePermission,
+	Description:              "Change the nickname of the specified user.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionUser,
+			Name:        "user",
+			Description: "User whose nickname to rename",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "nickname",
+			Description: "Nickname to rename user to",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "reason",
+			Description: "Reason for nickname change",
+			Required:    true,
+		},
+	},
+}
+
+var SlowmodeCommand = &discordgo.ApplicationCommand{
+	Name:                     "slowmode",
+	DefaultMemberPermissions: &channelPermission,
+	Description:              "Add slowmode to current channel.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionString,
+			Name:        "duration",
+			Description: "Duration of slowmode (e.g \"1m\", \"1h\", \"1d\")",
+			Required:    true,
+		},
+	},
+}
+
+var SlowmodeOffCommand = &discordgo.ApplicationCommand{
+	Name:                     "slowmodeoff",
+	DefaultMemberPermissions: &channelPermission,
+	Description:              "Remove slowmode from current channel.",
+}
+
+var PurgeCommand = &discordgo.ApplicationCommand{
+	Name:                     "purge",
+	DefaultMemberPermissions: &messagePermission,
+	Description:              "Delete a number of messages from a channel.",
+	Options: []*discordgo.ApplicationCommandOption{
+		{
+			Type:        discordgo.ApplicationCommandOptionChannel,
+			Name:        "channel",
+			Description: "Channel to purge",
+			Required:    true,
+		},
+		{
+			Type:        discordgo.ApplicationCommandOptionInteger,
+			Name:        "message-number",
+			Description: "Number of messages to purge (100 max)",
+			MaxValue:    100,
```

Comment:
I believe Discord gets mad if we delete a very large number of messages. There's also a [bulk delete function](https://pkg.go.dev/github.com/bwmarrin/discordgo#Session.ChannelMessagesBulkDelete) in discordgo which has a limit of 100, so this number was based off that.

Reply:
technically we can raise this limit by sleeping per 100 messages. we could potentially make a new command where it's truly bulk delete but then warn the user that it will take a very long time to purge an entire channel

Reply:
as discussed in discord if the # of messages is <= 100, don't worry about sleeping. otherwise set a sleep timer to delete every `n messages % 100` times

Reply:
closing this so the PR goes in but let's keep track of this :3